### PR TITLE
chore(config)!: Default `expire_internal_metrics` to 300s

### PR DESCRIPTION
**Pull request by jszwedko**
See #20562 for discussion.

Closes: #20562

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

**Comment by jcmcken** [comment]
I don't have this use case, but if someone wants to retain the previous behavior, is that possible with this change? Just for backwards compatibility. Or is the solution just to set it to a very high value?

I'm OK either way, just figured it was worth thinking about.

**Comment by jszwedko** [comment]
> I don't have this use case, but if someone wants to retain the previous behavior, is that possible with this change? Just for backwards compatibility. Or is the solution just to set it to a very high value?
> 
> I'm OK either way, just figured it was worth thinking about.

Good thought: ideally it could be set to `null`, but given that `null` isn't a valid TOML value, instead maybe we should interpret any negative as "don't expire".

File: lib/vector-core/src/config/global_options.rs [hunk 1]
```diff
@@ -353,7 +359,7 @@ mod tests {
         dd2: Option<P>,
         result: impl Fn(GlobalOptions) -> T,
     ) -> Result<T, Vec<String>> {
-        // Use TOML parsing to match the behavior of what a user would actually configure.
+        //// Use TOML parsing to match the behavior of what a user would actually configure.
```
Comment (pront) [inline lib/vector-core/src/config/global_options.rs:362]: nit: remove extra `//`

**Review by pront — COMMENTED** [review]
Pushed a commit to fix some errors; https://github.com/vectordotdev/vector/pull/20710/commits/598f166c76d4be160790ce9422782d9c78279222

File: lib/vector-core/src/config/global_options.rs (hunk 1 above)
Comment (jszwedko) [inline lib/vector-core/src/config/global_options.rs:362]:
```suggestion
        // Use TOML parsing to match the behavior of what a user would actually configure.
```

File: changelog.d/expire_internal_metrics.breaking.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+The global `expire_metrics_secs` configuration option now defaults to 300s rather than being
+disabled. Set to a negative value to disable expiration.
```
Comment (pront) [inline changelog.d/expire_internal_metrics.breaking.md:2]: > Set to a negative value to disable expiration.
Maybe worth adding to the note, that setting this to a negative value will preserve the existing (old) behavior.

**Review by pront — APPROVED** [review]

File: changelog.d/expire_internal_metrics.breaking.md (hunk 2 above)
Comment (jszwedko) [reply changelog.d/expire_internal_metrics.breaking.md]: Good call 👍

**Pull request merged**